### PR TITLE
Update for go modules

### PR DESCRIPTION
**Pull request by amanbolat**
I used some new not yet merged code from golang repo to make this library work with go modules.

Link to the source: https://go-review.googlesource.com/c/tools/+/126535

### CODE REVIEW

**Comment by sgtsquiggs** [comment]
Love to see this merged in; otherwise currently I have to fake a GOPATH to use enumer

**Comment by dmarkham** [comment]
I have merged this in my new and supported version that is a drop in replacement.  If your interested.
https://github.com/dmarkham/enumer

File: stringer.go [hunk 1]
```diff
@@ -434,35 +472,35 @@ func (f *File) genDecl(node ast.Node) bool {
 		// We now have a list of names (from one line of source code) all being
 		// declared with the desired type.
 		// Grab their names and actual values and store them in f.values.
-		for _, name := range vspec.Names {
-			if name.Name == "_" {
+		for _, n := range vspec.Names {
```
Comment (alvaroloes) [inline stringer.go]: The scope of the variable `n` is sufficiently large to deserve a longer name. I would stick with the previous name `name`. What was wrong with it?

File: stringer.go [hunk 2]
```diff
@@ -495,13 +533,13 @@ func usize(n int) int {
 func (g *Generator) declareIndexAndNameVars(runs [][]Value, typeName string) {
 	var indexes, names []string
 	for i, run := range runs {
-		index, name := g.createIndexAndNameDecl(run, typeName, fmt.Sprintf("_%d", i))
+		index, n := g.createIndexAndNameDecl(run, typeName, fmt.Sprintf("_%d", i))
```
Comment (alvaroloes) [inline stringer.go]: Same here and in other places where `name` was replaced by `n`

File: stringer.go [hunk 3]
```diff
@@ -529,14 +567,14 @@ func (g *Generator) createIndexAndNameDecl(run []Value, typeName string, suffix
...
 	for i, v := range indexes {
 		if i > 0 {
-			fmt.Fprintf(b, ", ")
+			_, _ = fmt.Fprintf(b, ", ")
 		}
-		fmt.Fprintf(b, "%d", v)
+		_, _ = fmt.Fprintf(b, "%d", v)
 	}
-	fmt.Fprintf(b, "}")
+	_, _ = fmt.Fprintf(b, "}")
```
Comment (alvaroloes) [inline stringer.go]: Definitely not worth this change. Could you please remove the changes you did to include `_, _ = ` ?

File: stringer.go [hunk 4]
```diff
@@ -138,16 +139,18 @@ type Generator struct {
...
 type File struct {
 	pkg  *Package  // Package to which this file belongs.
 	file *ast.File // Parsed AST.
 	// These fields are reset for each type being generated.
-	typeName string  // Name of the constant type.
-	values   []Value // Accumulator for constant values of that type.
+	typeName    string  // Name of the constant type.
+	values      []Value // Accumulator for constant values of that type.
+	trimPrefix  string
+	lineComment bool
```
Comment (alvaroloes) [inline stringer.go]: The two new fields added `trimPrefix` and `lineComment` doesn't seems to be used. Could you please remove it?

File: stringer.go [hunk 5]
```diff
@@ -159,76 +162,111 @@ type Package struct {
...
+// 	}
+// 	if len(astFiles) == 0 {
+// 		log.Fatalf("%s: no buildable Go files", directory)
+// 	}
+// 	g.pkg.name = astFiles[0].Name.Name
+// 	g.pkg.files = files
+// 	g.pkg.dir = directory
+// 	// Type check the package.
+// 	g.pkg.check(fs, astFiles)
+// }
```
Comment (alvaroloes) [inline stringer.go]: Could you please remove all the commented code?

**Comment by alvaroloes** [comment]
This should be fixed in the latest version. Closing. Thanks a lot!

**Pull request closed without merging**